js/source/geojson_source.js

```diff
@@ -202,31 +166,24 @@ GeoJSONSource.prototype = util.inherit(Evented, /** @lends GeoJSONSource.prototy
                 tile.redoPlacement(this);
             }
 
-            this.fire('tile.load', {tile: tile});
+            return callback(null, data);
 
         }.bind(this), this.workerID);
     },
 
-    _abortTile: function(tile) {
+    abort: function(tile) {
         tile.aborted = true;
     },
 
-    _addTile: function(tile) {
-        this.fire('tile.add', {tile: tile});
-    },
-
-    _removeTile: function(tile) {
-        this.fire('tile.remove', {tile: tile});
-    },
-
-    _unloadTile: function(tile) {
+    unload: function(tile) {
         tile.unloadVectorData(this.map.painter);
         this.dispatcher.send('remove tile', { uid: tile.uid, source: this.id }, null, tile.workerID);
     },
 
-    redoPlacement: Source.redoPlacement,
-
-    _redoTilePlacement: function(tile) {
-        tile.redoPlacement(this);
+    serialize: function() {
+        return {
+            type: this.type,
+            data: this._data
+        };
     }
-});
+};
```

Comment:
@lucaswoj Yeah, this should be made consistent... 

> Isn't this whole class private?

Hmm -`minzoom`, `maxzoom`, `load`, `serialize`, etc. are implementing the common "Source" interface that we're trying to establish in this PR.  `setData` is a user-facing method.  It seems to me that these should be public... am I misunderstanding your question?